pkg/test/discovery_client.go

```diff
@@ -76,7 +73,7 @@ func (c *DiscoveryClient) WithAPIResource(resource *APIResource) *DiscoveryClien
 		Namespaced:   resource.Namespaced,
 		Group:        resource.Group,
 		Version:      resource.Version,
-		Kind:         cases.Title(language.Und).String(strings.TrimSuffix(resource.Name, "s")),
```

Comment:
The conversion code here is incorrect. We expect to convert `persistentvolumes` to `PersistentVolume`, but actually get `Persistentvolume`.

This causes unit test failure mentioned in https://github.com/vmware-tanzu/velero/pull/7322